src/SecTester.Repeater/Bus/Formatters/MessagePackHttpHeadersFormatter.cs

```diff
@@ -0,0 +1,161 @@
+    MessagePackSerializerOptions options)
+  {
+    if (value == null)
+    {
+      writer.WriteNil();
+    }
+    else
+    {
+      var count = value.Count();
+
+      writer.WriteMapHeader(count);
+
+      SerializeMap(ref writer, value);
+    }
+  }
+
+  public IEnumerable<KeyValuePair<string, IEnumerable<string>>>? Deserialize(ref MessagePackReader reader,
+    MessagePackSerializerOptions options)
+  {
+    switch (reader.NextMessagePackType)
+    {
+      case MessagePackType.Nil:
+        return null;
+      case MessagePackType.Map:
+        break;
+      default:
+        throw new MessagePackSerializationException(string.Format(CultureInfo.InvariantCulture,
+          "Unrecognized code: 0x{0:X2} but expected to be a map or null", reader.NextCode));
+    }
+
+    var length = reader.ReadMapHeader();
+
+    options.Security.DepthStep(ref reader);
+
+    try
+    {
+      return DeserializeMap(ref reader, length, options);
+    }
+    finally
+    {
+      reader.Depth--;
+    }
+  }
+
+  private static void SerializeMap(ref MessagePackWriter writer, IEnumerable<KeyValuePair<string, IEnumerable<string>>> value)
+  {
+    foreach (var item in value)
+    {
+      writer.Write(item.Key);
+
+      SerializeValue(ref writer, item);
+    }
+  }
+
+  private static void SerializeValue(ref MessagePackWriter writer, KeyValuePair<string, IEnumerable<string>> item)
+  {
+    var headersCount = item.Value.Count();
+
+    if (headersCount == 1)
+    {
+      writer.Write(item.Value.First());
+    }
+    else
+    {
+      writer.WriteArrayHeader(headersCount);
+
+      foreach (var subItem in item.Value)
+      {
+        writer.Write(subItem);
+      }
+    }
+  }
+
+  private static List<KeyValuePair<string, IEnumerable<string>>> DeserializeMap(ref MessagePackReader reader, int length,
+    MessagePackSerializerOptions options)
+  {
+    var result = new List<KeyValuePair<string, IEnumerable<string>>>(length);
+
+    for (var i = 0 ; i < length ; i++)
+    {
+      var key = DeserializeString(ref reader);
+
+      switch (reader.NextMessagePackType)
+      {
+        case MessagePackType.String:
+          result.Add(new KeyValuePair<string, IEnumerable<string>>(key, new List<string> { DeserializeString(ref reader) }));
+          break;
+        case MessagePackType.Array:
+          result.Add(new KeyValuePair<string, IEnumerable<string>>(key, DeserializeArray(ref reader, reader.ReadArrayHeader(), options)));
+          break;
+        default:
+          throw new MessagePackSerializationException(string.Format(CultureInfo.InvariantCulture,
+            "Unrecognized code: 0x{0:X2} but expected to be either a string or an array.", reader.NextCode));
+      }
+    }
+
+
+    return result;
+  }
+
+  private static IEnumerable<string> DeserializeArray(ref MessagePackReader reader, int length, MessagePackSerializerOptions options)
+  {
+    var result = new List<string>(length);
+
+    if (length == 0)
+    {
+      return result;
+    }
```

Comment:
Avoid redundant conditions and branches; they only complicate the code w/o reason